DemoApp/Sources/Components/Stats/ParticipantStatsView.swift

```diff
@@ -11,91 +11,114 @@ struct ParticipantStatsView: View {
+                            Text("Fetching stats...")
+                                .font(valueFont)
+                            ProgressView()
+                        }
+                    }
+                } else {
+                    ForEach(viewModel.statsEntries) { entry in
+                        tileView {
+                            VStack(alignment: .center, spacing: spacing) {
+                                Spacer()
+                                Text(entry.title)
+                                    .font(fonts.caption1)
+                                    .lineLimit(1)
+                                Text(entry.value)
+                                    .font(valueFont)
+                                Spacer()
+                            }
+                        }
+                    }
+
+                    tileView {
+                        Button {
+                            viewModel.allStatsShown = true
+                        } label: {
+                            Text("All stats")
+                                .font(valueFont)
+                        }
+                    }
                 }
-                .frame(width: itemSize, height: itemSize / 2)
-                .padding(.all, spacing)
-                .background(Color(colors.background1))
-                .cornerRadius(8)
-            }
-            Button {
-                viewModel.allStatsShown = true
-            } label: {
-                Text("All stats")
-                    .font(valueFont)
-                    .foregroundColor(colors.text)
-                    .frame(width: itemSize, height: itemSize / 2)
-                    .padding(.all, spacing)
-                    .background(Color(colors.background1))
-                    .cornerRadius(8)
             }
-
         }
-        .frame(width: 2 * itemSize + 4 * spacing)
-        .padding()
+        .navigationTitle("Stats")
         .background(colors.callControlsBackground)
-        .clipped()
-        .clipShape(RoundedRectangle(cornerRadius: 8))
-        .opacity(viewModel.statsEntries.count > 0 ? 1 : 0)
         .sheet(
             isPresented: $viewModel.allStatsShown,
             content: {
                 RawStatsView(statsReport: viewModel.statsReport)
             }
         )
+        .toolbar(content: {
+            ToolbarItem(placement: .navigationBarLeading) {
+                Button {
+                    presentationBinding.wrappedValue = false
+                } label: {
+                    Text("Close")
+                        .foregroundColor(colors.text)
+                }
+            }
+        })
     }
-    
-    private var itemSize: CGFloat {
-        if compacted {
-            return 72
-        } else {
-            return 100
-        }
+
+    @ViewBuilder
+    private func tileView(@ViewBuilder _ content: () -> some View) -> some View {
+        content()
+            .frame(height: max(floor(itemSize.height), 50))
```

Comment:
why do we do this?

Reply:
The floor is because if the value has a floating point renderer will round it up, causing the value to go higher than the available space and the LazyVStack will break the layout.

The max is to solve an issue where the tiles weren't getting a consistent size but that may be gone now. Will try